Emby.Server.Implementations/HttpServer/HttpResultFactory.cs

```diff
@@ -595,6 +589,27 @@ public async Task<object> GetStaticResult(IRequest requestContext, StaticResultO
                 }
             }
 
+            byte[] hash;
+            // Don't md5 the stream if it's over 10 MB, it's just not worth it
+            if (totalContentLength.HasValue && totalContentLength.Value < 10000000)
+            {
+                // Generate an etag based on stream content
+                hash = MD5.Create().ComputeHash(stream);
+                // reset position so the response can re-use it
+                stream.Position = 0;
+            }
+            else
+            {
+                var hashId = options.Path + totalContentLength + options.ContentType;
```

Comment:
This could use at least file modification date, plus maybe first 1mb of file contents. Just to be on the safe side.